src/compas/datastructures/tree/tree.py

```diff
@@ -0,0 +1,259 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import annotations
+
+from compas.datastructures import Datastructure
+
+
+class TreeNode(object):
+    """A node of a tree data structure.
+
+    Parameters
+    ----------
+    name: str, optional
+        The name of the tree ndoe.
+    attributes: dict[str, Any], optional
+        User-defined attributes of the datastructure.
+
+    Attributes
+    ----------
+    name: str
+        The name of the datastructure.
+    attributes : dict[str, Any]
+        User-defined attributes of the datastructure.
+    parent: :class:`~compas.datastructures.TreeNode`
+        The parent node of the tree node.
+    children: set[:class:`~compas.datastructures.TreeNode`]
+        The children of the tree node.
+    tree: :class:`~compas.datastructures.Tree`
+        The tree the node belongs to.
+    is_root: bool
+        True if the node is the root node of the tree.
+    is_leaf: bool
+        True if the node is a leaf node of the tree.
+    is_branch: bool
+        True if the node is a branch node of the tree.
+    acestors: generator[:class:`~compas.datastructures.TreeNode`]
+        A generator of the acestors of the tree node.
+    descendants: generator[:class:`~compas.datastructures.TreeNode`]
+        A generator of the descendants of the tree node.
+
+    """
+
+    def __init__(self, name: str = None, attributes: dict = None):
```

Comment:
you can't use typing in the core compas lib...

Reply:
removed them